File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for ch in s:
        if len(stack) == 0:
            stack.append(ch)
            continue
        
        if stack[-1] == '(' and ch == ')':
            stack.pop()
        else:
            stack.append(ch)
    if stack:
        return False
    else:
        return True
            

File: C++/프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <vector>

using namespace std;

bool solution(string s){
    vector<char> st;
    
    for(char c : s){
        // 비어있다면 그냥 넣음
        if(st.empty()){
            st.push_back(c);
        }else{
            char back = st.back();
            // pop할 조건
            if(back == '(' && c == ')'){
                st.pop_back();
            }else{
                st.push_back(c);
            }
        }
    }
    
    return st.empty();
    
}